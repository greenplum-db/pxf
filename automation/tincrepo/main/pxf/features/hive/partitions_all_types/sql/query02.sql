-- @description query02 for PXF Hive table with multiple partitions - queries for null values
SET datestyle TO 'ISO, MDY';

SELECT * FROM hive_partitions_all_types WHERE t2 is null ORDER BY t1;

SELECT * FROM hive_partitions_all_types WHERE num1 is null ORDER BY t1;

SELECT * FROM hive_partitions_all_types WHERE dub1 is null ORDER BY t1;

SELECT * FROM hive_partitions_all_types WHERE dec1 is null ORDER BY t1;

SELECT * FROM hive_partitions_all_types WHERE tm is null ORDER BY t1;

SELECT * FROM hive_partitions_all_types WHERE r is null ORDER BY t1;

SELECT * FROM hive_partitions_all_types WHERE bg is null ORDER BY t1;

SELECT * FROM hive_partitions_all_types WHERE b is null ORDER BY t1;

SELECT * FROM hive_partitions_all_types WHERE tn is null ORDER BY t1;

SELECT * FROM hive_partitions_all_types WHERE sml is null ORDER BY t1;

SELECT * FROM hive_partitions_all_types WHERE dt is null ORDER BY t1;

SELECT * FROM hive_partitions_all_types WHERE vc1 is null ORDER BY t1;

SELECT * FROM hive_partitions_all_types WHERE c1 is null ORDER BY t1;

SELECT * FROM hive_partitions_all_types WHERE bin is null ORDER BY t1;
