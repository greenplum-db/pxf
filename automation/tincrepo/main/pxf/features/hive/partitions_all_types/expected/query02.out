-- @description query02 for PXF Hive table with multiple partitions - queries for null values
SET datestyle TO 'ISO, MDY';
SELECT * FROM hive_partitions_all_types WHERE t2 is null ORDER BY t1;
       t1        | t2 | num1 | dub1 |         dec1         |         tm          |  r  |    bg    | b | tn | sml  |     dt     |  vc1  | c1  | bin 
-----------------+----+------+------+----------------------+---------------------+-----+----------+---+----+------+------------+-------+-----+-----
 row12_text_null |    |   11 |   37 | 0.123456789012345679 | 2013-07-23 21:00:05 | 7.7 | 23456789 | f | 11 | 1100 | 2015-03-06 | abcde | ab  | 31
(1 row)

SELECT * FROM hive_partitions_all_types WHERE num1 is null ORDER BY t1;
       t1       |  t2  | num1 | dub1 |         dec1         |         tm          |  r  |    bg    | b | tn | sml  |     dt     |  vc1  | c1  | bin 
----------------+------+------+------+----------------------+---------------------+-----+----------+---+----+------+------------+-------+-----+-----
 row13_int_null | s_16 |      |   37 | 0.123456789012345679 | 2013-07-23 21:00:05 | 7.7 | 23456789 | f | 11 | 1100 | 2015-03-06 | abcde | ab  | 31
(1 row)

SELECT * FROM hive_partitions_all_types WHERE dub1 is null ORDER BY t1;
        t1         |  t2  | num1 | dub1 |         dec1         |         tm          |  r  |    bg    | b | tn | sml  |     dt     |  vc1  | c1  | bin 
-------------------+------+------+------+----------------------+---------------------+-----+----------+---+----+------+------------+-------+-----+-----
 row14_double_null | s_16 |   11 |      | 0.123456789012345679 | 2013-07-23 21:00:05 | 7.7 | 23456789 | f | 11 | 1100 | 2015-03-06 | abcde | ab  | 31
(1 row)

SELECT * FROM hive_partitions_all_types WHERE dec1 is null ORDER BY t1;
         t1         |  t2  | num1 | dub1 | dec1 |         tm          |  r  |    bg    | b | tn | sml  |     dt     |  vc1  | c1  | bin 
--------------------+------+------+------+------+---------------------+-----+----------+---+----+------+------------+-------+-----+-----
 row15_decimal_null | s_17 |   12 |   38 |      | 2013-07-24 21:00:05 | 7.7 | 23456789 | f | 11 | 1100 | 2015-03-06 | abcde | ab  | 31
(1 row)

SELECT * FROM hive_partitions_all_types WHERE tm is null ORDER BY t1;
          t1          |  t2  | num1 | dub1 |         dec1         | tm |  r  |    bg    | b | tn | sml  |     dt     |  vc1  | c1  | bin 
----------------------+------+------+------+----------------------+----+-----+----------+---+----+------+------------+-------+-----+-----
 row16_timestamp_null | s_16 |   11 |   37 | 0.123456789012345679 |    | 7.7 | 23456789 | f | 11 | 1100 | 2015-03-06 | abcde | ab  | 31
(1 row)

SELECT * FROM hive_partitions_all_types WHERE r is null ORDER BY t1;
       t1        |  t2  | num1 | dub1 |         dec1         |         tm          | r |    bg    | b | tn | sml  |     dt     |  vc1  | c1  | bin 
-----------------+------+------+------+----------------------+---------------------+---+----------+---+----+------+------------+-------+-----+-----
 row17_real_null | s_16 |   11 |   37 | 0.123456789012345679 | 2013-07-23 21:00:05 |   | 23456789 | f | 11 | 1100 | 2015-03-06 | abcde | ab  | 31
(1 row)

SELECT * FROM hive_partitions_all_types WHERE bg is null ORDER BY t1;
        t1         |  t2  | num1 | dub1 |         dec1         |         tm          |  r  | bg | b | tn | sml  |     dt     |  vc1  | c1  | bin 
-------------------+------+------+------+----------------------+---------------------+-----+----+---+----+------+------------+-------+-----+-----
 row18_bigint_null | s_16 |   11 |   37 | 0.123456789012345679 | 2013-07-23 21:00:05 | 7.7 |    | f | 11 | 1100 | 2015-03-06 | abcde | ab  | 31
(1 row)

SELECT * FROM hive_partitions_all_types WHERE b is null ORDER BY t1;
       t1        |  t2  | num1 | dub1 |         dec1         |         tm          |  r  |    bg    | b | tn | sml  |     dt     |  vc1  | c1  | bin 
-----------------+------+------+------+----------------------+---------------------+-----+----------+---+----+------+------------+-------+-----+-----
 row19_bool_null | s_16 |   11 |   37 | 0.123456789012345679 | 2013-07-23 21:00:05 | 7.7 | 23456789 |   | 11 | 1100 | 2015-03-06 | abcde | ab  | 31
(1 row)

SELECT * FROM hive_partitions_all_types WHERE tn is null ORDER BY t1;
         t1         |  t2  | num1 | dub1 |         dec1         |         tm          |  r  |    bg    | b | tn | sml  |     dt     |  vc1  | c1  | bin 
--------------------+------+------+------+----------------------+---------------------+-----+----------+---+----+------+------------+-------+-----+-----
 row20_tinyint_null | s_16 |   11 |   37 | 0.123456789012345679 | 2013-07-23 21:00:05 | 7.7 | 23456789 | f |    | 1100 | 2015-03-06 | abcde | ab  | 31
(1 row)

SELECT * FROM hive_partitions_all_types WHERE sml is null ORDER BY t1;
         t1          |  t2  | num1 | dub1 |         dec1         |         tm          |  r  |    bg    | b | tn | sml |     dt     |  vc1  | c1  | bin 
---------------------+------+------+------+----------------------+---------------------+-----+----------+---+----+-----+------------+-------+-----+-----
 row21_smallint_null | s_16 |   11 |   37 | 0.123456789012345679 | 2013-07-23 21:00:05 | 7.7 | 23456789 | f | 11 |     | 2015-03-06 | abcde | ab  | 31
(1 row)

SELECT * FROM hive_partitions_all_types WHERE dt is null ORDER BY t1;
       t1        |  t2  | num1 | dub1 |         dec1         |         tm          |  r  |    bg    | b | tn | sml  | dt |  vc1  | c1  | bin 
-----------------+------+------+------+----------------------+---------------------+-----+----------+---+----+------+----+-------+-----+-----
 row22_date_null | s_16 |   11 |   37 | 0.123456789012345679 | 2013-07-23 21:00:05 | 7.7 | 23456789 | f | 11 | 1100 |    | abcde | ab  | 31
(1 row)

SELECT * FROM hive_partitions_all_types WHERE vc1 is null ORDER BY t1;
         t1         |  t2  | num1 | dub1 |         dec1         |         tm          |  r  |    bg    | b | tn | sml  |     dt     | vc1 | c1  | bin 
--------------------+------+------+------+----------------------+---------------------+-----+----------+---+----+------+------------+-----+-----+-----
 row23_varchar_null | s_16 |   11 |   37 | 0.123456789012345679 | 2013-07-23 21:00:05 | 7.7 | 23456789 | f | 11 | 1100 | 2015-03-06 |     | ab  | 31
(1 row)

SELECT * FROM hive_partitions_all_types WHERE c1 is null ORDER BY t1;
       t1        |  t2  | num1 | dub1 |         dec1         |         tm          |  r  |    bg    | b | tn | sml  |     dt     |  vc1  | c1 | bin 
-----------------+------+------+------+----------------------+---------------------+-----+----------+---+----+------+------------+-------+----+-----
 row24_char_null | s_16 |   11 |   37 | 0.123456789012345679 | 2013-07-23 21:00:05 | 7.7 | 23456789 | f | 11 | 1100 | 2015-03-06 | abcde |    | 31
(1 row)

SELECT * FROM hive_partitions_all_types WHERE bin is null ORDER BY t1;
        t1         |  t2  | num1 | dub1 |         dec1         |         tm          |  r  |    bg    | b | tn | sml  |     dt     |  vc1  | c1  | bin 
-------------------+------+------+------+----------------------+---------------------+-----+----------+---+----+------+------------+-------+-----+-----
 row25_binary_null | s_16 |   11 |   37 | 0.123456789012345679 | 2013-07-23 21:00:05 | 7.7 | 23456789 | f |  1 | 1100 | 2015-03-06 | abcde | ab  | 
(1 row)

